src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,155 @@
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+    }
+
+    @Test
+    void noId_nullId_notOk() {
+        User user = new User(null, "correctLogin", "correctPassword", 18);
+        ValidationException actual = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals(ErrorMessage.ERROR_ID_REQUIRED, actual.getMessage());
+    }
+
+    @Test
+    void noLogin_nullLogin_notOk() {
+        User user = new User(123L, null, "correctPassword", 18);
+        ValidationException actual = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals(ErrorMessage.ERROR_LOGIN_REQUIRED, actual.getMessage());
+
+    }
+
+    @Test
+    void noPassword_nullPassword_notOk() {
+        User user = new User(123L, "correctLogin", null, 18);
+        ValidationException actual = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals(ErrorMessage.ERROR_PASSWORD_REQUIRED, actual.getMessage());
+
+    }
+
+    @Test
+    void noAge_nullAge_notOk() {
+        User user = new User(123L, "correctLogin", "correctPassword", null);
+        ValidationException actual = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals(ErrorMessage.ERROR_AGE_REQUIRED, actual.getMessage());
+    }
+
+    @Test
+    void userAge_under18_notOk() {
+        User user = new User(123L, "correctLogin", "correctPassword", 17);
+        ValidationException actual = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals(ErrorMessage.ERROR_UNDERAGE_USER, actual.getMessage());
+    }
+
+    @Test
+    void userAge_18_Ok() {
+        User user = new User(123L, "correctLogin", "correctPassword", 18);
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+    }
+
+    @Test
+    void userAge_22_Ok() {
+        User user = new User(123L, "correctLogin", "correctPassword", 22);
+        User actual = registrationService.register(user);
+        assertEquals(user, actual);
+    }
+
+    @Test
+    void negativeAge_minus10_notOk() {
+        User user = new User(123L, "correctLogin", "correctPassword", -10);
+        ValidationException actual = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals(ErrorMessage.ERROR_UNDERAGE_USER, actual.getMessage());
+    }
+
+    @Test
+    void spaceInLogin_spaceLogin_notOk() {
+        User user = new User(123L, "space Login", "correctPassword", 20);
+        ValidationException actual = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals(ErrorMessage.ERROR_SPACE_IN_LOGIN, actual.getMessage());
```

Comment:
Lots of unit tests in this class use the same logic - how about extracting it to a private method to simplify the code? Something like:
```java
private void assertUserRegistrationThrowsValidationExceptionWithMessage(User user, String errorMessage);
```